relay-server/src/actors/project.rs

```diff
@@ -939,7 +947,10 @@ impl Project {
         mut envelope: Box<Envelope>,
         mut envelope_context: EnvelopeContext,
     ) -> Result<CheckedEnvelope, DiscardReason> {
-        let state = self.valid_state();
+        // Treat invalid state as no state at all.
```

Comment:
_nit_: Create a doc-comment for `check_envelope` clearly mentioning: If you call this on an invalid project state, this will skip certain checks. This is super important, because it means you **must not** call this in the main processing path.